src/services/websocket_service.ts

```diff
@@ -56,70 +51,24 @@ export class WebsocketService {
-            for (const nested of decodedAssetData.nestedAssetData) {
-                data = [...data, ...WebsocketService._decodedContractAndAssetData(nested).data];
-            }
-        } else if (orderUtils.isStaticCallAssetData(decodedAssetData)) {
-            // do nothing
-        } else {
-            data = [
-                ...data,
-                // tslint:disable-next-line:no-unnecessary-type-assertion
-                (decodedAssetData as Exclude<typeof decodedAssetData, MultiAssetDataWithRecursiveDecoding>)
-                    .tokenAddress,
-            ];
-        }
-        return { data, assetProxyId: decodedAssetData.assetProxyId };
-    }
+    private _orderEventsSubscription?: ZenObservable.Subscription;
+    // TODO(kimpers): [V4] Do we want any other matching than takerToken and makerToken?
     private static _matchesOrdersChannelSubscription(
-        order: SignedOrder,
+        order: SignedLimitOrder,
         opts: OrdersChannelSubscriptionOpts | ALL_SUBSCRIPTION_OPTS,
     ): boolean {
         if (opts === 'ALL_SUBSCRIPTION_OPTS') {
             return true;
         }
-        const { makerAssetData, takerAssetData } = order;
-        const makerAssetDataTakerAssetData = [makerAssetData, takerAssetData];
-        // Handle the specific, unambiguous asset datas
-        // traderAssetData?: string;
-        if (opts.traderAssetData && makerAssetDataTakerAssetData.includes(opts.traderAssetData)) {
-            return true;
-        }
-        // makerAssetData?: string;
-        // takerAssetData?: string;
-        if (
-            opts.makerAssetData &&
-            opts.takerAssetData &&
-            makerAssetDataTakerAssetData.includes(opts.makerAssetData) &&
-            makerAssetDataTakerAssetData.includes(opts.takerAssetData)
-        ) {
-            return true;
-        }
-        // makerAssetAddress?: string;
-        // takerAssetAddress?: string;
-        const makerContractAndAssetData = WebsocketService._decodedContractAndAssetData(makerAssetData);
-        const takerContractAndAssetData = WebsocketService._decodedContractAndAssetData(takerAssetData);
+        const { makerToken, takerToken } = order;
+
         if (
-            opts.makerAssetAddress &&
-            opts.takerAssetAddress &&
-            makerContractAndAssetData.assetProxyId !== AssetProxyId.MultiAsset &&
-            makerContractAndAssetData.assetProxyId !== AssetProxyId.StaticCall &&
-            takerContractAndAssetData.assetProxyId !== AssetProxyId.MultiAsset &&
-            takerContractAndAssetData.assetProxyId !== AssetProxyId.StaticCall &&
-            makerContractAndAssetData.data.includes(opts.makerAssetAddress) &&
-            takerContractAndAssetData.data.includes(opts.takerAssetAddress)
+            (opts.takerToken && takerToken.toLowerCase() === opts.takerToken.toLowerCase()) ||
+            (opts.makerToken && makerToken.toLowerCase() === opts.makerToken.toLowerCase())
```

Comment:
Do we want to support any additional filters here?

Reply:
Nah I think it's fine.